transport_plugins/bled112/iotile_transport_bled112/virtual_bled112.py

```diff
@@ -193,22 +196,56 @@ def _queue_reports(self, *reports):
 
             self.reports.put(report)
 
-    def _advertisement(self):
+    def _advertisement(self, pkt_type=1):
         # Flags are
         # bit 0: whether we have pending data
         # bit 1: whether we are in a low voltage state
         # bit 2: whether another user is connected
         # bit 3: whether we support robust reports
         # bit 4: whether we allow fast writes
+        if self.AdvertisingPacketVer:
+            pkt_type = self.AdvertisingPacketVer
 
-        flags = (0 << 1) | (0 << 2) | (1 << 3) | (1 << 4) | (int(self.device.pending_data))
         ble_flags = struct.pack("<BBB", 2, 1, 0x4 | 0x2)  # General discoverability and no BR/EDR support
-        uuid_list = struct.pack("<BB16s", 17, 6, TileBusService.bytes_le)
-        manu = struct.pack("<BBHLH", 9, 0xFF, ArchManuID, self.device.iotile_id, flags)
+        flags = (0 << 1) | (0 << 2) | (1 << 3) | (1 << 4) | (int(self.device.pending_data))
+
+        if pkt_type == 1:
+            uuid_list = struct.pack("<BB16s", 17, 6, TileBusService.bytes_le)
+            manu = struct.pack("<BBHLH", 9, 0xFF, ArchManuID, self.device.iotile_id, flags)
+            return ble_flags + uuid_list + manu
+
+        elif pkt_type == 2:
+
+            self._logger.debug("PacketType:2 - {0}".format(self.json_args))
+
+            reboots = 123456
+            timestamp = calendar.timegm(time.gmtime())
+            voltage = 0x88 
+            OTHER = 0  #TODO
+            subsecond_cnt = 0xF
+            reserved = 0
 
-        return ble_flags + uuid_list + manu
+            bcast_stream = 1111
+            bcast_value = 22222222
+            mac = 33333333
+
+            reboots_hi = (reboots & 0xFF0000) >> 16
+            reboots_lo = (reboots & 0x00FFFF)
+
+            data1 = struct.pack("<BBHL", 27, 0x16, 0x03C0, self.device.iotile_id)
+            data2 = struct.pack("<HBBLBB", reboots_lo, reboots_hi, flags, timestamp, voltage, OTHER)
+            data3 = struct.pack("<HLL", bcast_stream, bcast_value, mac)
+
+            return ble_flags + data1 + data2 + data3
+
+        else:
+            self._logger.error("Invalid Advertising Packet Version")
```

Comment:
You should raise an exception here rather than that returning `None`.  The callers of this function probably cannot handle a `None` return value so it will just mask the source of the error.